lib/shoryuken/middleware/server/exponential_backoff_retry.rb

```diff
@@ -0,0 +1,54 @@
+module Shoryuken
+  module Middleware
+    module Server
+      class ExponentialBackoffRetry
+        include Util
+
+        def call(worker, queue, sqs_msg, body)
+          started_at = Time.now
+          yield
+        rescue Exception => e
+          retry_intervals = Array(worker.class.get_shoryuken_options['retry_intervals'])
+           
+          if retry_intervals.empty? || ! handle_failure(sqs_msg, started_at, retry_intervals)
+            # Re-raise the exception if the job is not going to be retried.
+            # This allows custom middleware (like exception notifiers) to be aware of the unhandled failure.
+            raise e
+          end
+        end
+        
+      private
+      
+        def handle_failure(sqs_msg, started_at, retry_intervals)
+          attempts = sqs_msg.attributes['ApproximateReceiveCount']
```

Comment:
hm actually I believe we don't need `||=` with `%[All]`,  because `All` will cause to return everything.